src/test/scala/org/apache/spark/sql/delta/DeltaSourceSuite.scala

```diff
@@ -964,6 +964,71 @@ class DeltaSourceSuite extends DeltaSourceSuiteBase {
     }
   }
 
+  test("Read data using the .delta() method extended from the DataFrameReader API") {
+    withTempDir { tempDir =>
+      Seq(1, 2, 3).toDF("x").write.format("delta").save(tempDir.getCanonicalPath)
+
+      val df = spark.read.delta(tempDir.toString)
+      val expectedRows = Seq(Row(1), Row(2), Row(3))
+      checkAnswer(df, expectedRows)
+    }
+  }
+
+  test("Write data using the .delta() method extended from the DataFrameWriter API") {
+    withTempDir { tempDir =>
+      Seq(1, 2, 3).toDF("x").write.delta(tempDir.getCanonicalPath)
+
+      val df = spark.read.format("delta").load(tempDir.toString)
+      val expectedRows = Seq(Row(1), Row(2), Row(3))
+      checkAnswer(df, expectedRows)
+    }
+  }
+
+  test("read stream using the .delta() method extended from the DataStreamReader API") {
+    withTempDir { inputDir =>
+      val deltaLog = DeltaLog.forTable(spark, new Path(inputDir.toURI))
+      withMetadata(deltaLog, StructType.fromDDL("value STRING"))
+
+      val df = spark.readStream
+        .delta(inputDir.getCanonicalPath)
+        .filter($"value" contains "keep")
```

Comment:
This `filter` is not needed here, is it?

Reply:
I have removed it. Thanks for your feedback @jaceklaskowski